solutions/4-getting-started-js/group_by.js

```diff
@@ -0,0 +1,31 @@
+// eslint-disable-next-line no-var
+var groupBy = function (groupingFunction, arr) {
+  const properties = new Set()
+  const result = {}
+  arr.forEach(item => properties.add(groupingFunction(item)))
+  properties.forEach(property =>
```

Comment:
Let's try to solve this using only JS objects and arrays, and not Set.

Also, you might want to look into the .reduce() function.